backend/module/proxy/proxy.go

```diff
@@ -23,23 +34,119 @@ func New(cfg *any.Any, log *zap.Logger, scope tally.Scope) (module.Module, error
 	}
 
 	m := &mod{
-		logger: log,
-		scope:  scope,
+		client:   &http.Client{},
+		services: config.Services,
+		logger:   log,
+		scope:    scope,
 	}
 
 	return m, nil
 }
 
 type mod struct {
-	logger *zap.Logger
-	scope  tally.Scope
+	client   HTTPClient
+	services []*proxyv1cfg.Service
+	logger   *zap.Logger
+	scope    tally.Scope
 }
 
 func (m *mod) Register(r module.Registrar) error {
 	proxyv1.RegisterProxyAPIServer(r.GRPCServer(), m)
 	return r.RegisterJSONGateway(proxyv1.RegisterProxyAPIHandler)
 }
 
-func (m *mod) RequestProxy(context.Context, *proxyv1.RequestProxyRequest) (*proxyv1.RequestProxyResponse, error) {
-	return nil, errors.New("not implemented")
+func (m *mod) RequestProxy(ctx context.Context, req *proxyv1.RequestProxyRequest) (*proxyv1.RequestProxyResponse, error) {
+	if !isAllowedRequest(m.services, req.Service, req.Path, req.HttpMethod) {
+		return nil, errors.New("This request is not allowed, check the proxy configuration.")
+	}
+
+	// If its allowed lookup the service
+	service := &proxyv1cfg.Service{}
+	for _, s := range m.services {
+		if s.Name == req.Service {
+			service = s
+		}
+	}
+
+	// Set all additional headers if specified
+	headers := http.Header{}
+	if len(service.Headers) > 0 {
+		for k, v := range service.Headers {
+			headers.Add(k, v)
+		}
+	}
+
+	// Parse the URL by joining both the HOST and PATH specifed by the config
+	parsedUrl, err := url.Parse(fmt.Sprintf("%s%s", service.Host, req.Path))
```

Comment:
should add validation on config in New or using validate to ensure that the host is generally correctly formed